libs/runtime/ebpf_epoch.c

```diff
@@ -44,16 +72,27 @@
  */
 typedef __declspec(align(EBPF_CACHE_LINE_SIZE)) struct _ebpf_epoch_cpu_entry
 {
-    LIST_ENTRY epoch_state_list;           ///< Per-CPU list of thread entries.
-    ebpf_list_entry_t free_list;           ///< Per-CPU free list.
-    int64_t current_epoch;                 ///< The current epoch for this CPU.
-    int64_t released_epoch;                ///< The newest epoch that can be released.
-    int timer_armed : 1;                   ///< Set if the flush timer is armed.
-    int rundown_in_progress : 1;           ///< Set if rundown is in progress.
-    int epoch_computation_in_progress : 1; ///< Set if epoch computation is in progress.
-    ebpf_timed_work_queue_t* work_queue;   ///< Work queue used to schedule work items.
+    LIST_ENTRY epoch_state_list;                ///< Per-CPU list of thread entries.
+    ebpf_list_entry_t free_list;                ///< Per-CPU free list.
+    int64_t current_epoch;                      ///< The current epoch for this CPU.
+    int64_t released_epoch;                     ///< The newest epoch that can be released.
+    uint64_t timer_armed : 1;                   ///< Set if the flush timer is armed.
+    uint64_t rundown_in_progress : 1;           ///< Set if rundown is in progress.
+    uint64_t epoch_computation_in_progress : 1; ///< Set if epoch computation is in progress.
+    uint64_t active : 1; ///< CPU is active in epoch computation. Only accessed under _ebpf_epoch_active_cpu_list_lock.
+    uint64_t work_queue_assigned : 1;    ///< Work queue is assigned to this CPU.
+    uint64_t reserved : 59;              ///< Reserved for future use.
+    ebpf_timed_work_queue_t* work_queue; ///< Work queue used to schedule work items.
 } ebpf_epoch_cpu_entry_t;
 
+static_assert(
+    sizeof(ebpf_epoch_cpu_entry_t) == EBPF_CACHE_LINE_SIZE, "ebpf_epoch_cpu_entry_t must fit within one cache line");
+
+/**
+ * @brief Lock to ensure a consistent view of the active CPUs.
+ */
+static ebpf_lock_t _ebpf_epoch_active_cpu_list_lock; ///< Lock to protect the active CPU list.
```

Comment:
Consider renaming as this protects the cpu entry table and not only the active cpu list.